kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeResultGenerator.kt

```diff
@@ -116,32 +116,50 @@ data class TypeResultGenerator(val options: Options, val implicitAccessor: Acces
                 add("%N", methodName)
             } else {
                 add("%N(", methodName)
-                parameters.forEachIndexed { i, param ->
+                if (parameters.isNotEmpty()) {
+                    add("\n⇥")
+                }
+                parameters.entries.forEachIndexed { i, (paramName, param) ->
                     if (i != 0) {
-                        add(",")
+                        add(",\n")
                     }
                     with(if (changeScope) copy(implicitAccessor = accessor) else this@TypeResultGenerator) {
+                        add("$paramName = ")
                         add(param.generate())
                     }
                 }
+                if (parameters.isNotEmpty()) {
+                    add("\n⇤")
+                }
                 add(")")
             }
 
             if (changeScope) {
-                add("\n")
-                endControlFlow()
+                // don't use endControlFlow() because it emits a newline after the closing brace
+                add("\n⇤}")
             }
         }.build()
     }
 
     private fun TypeResult.Constructor.generate(): CodeBlock {
         return CodeBlock.builder().apply {
             add("%T(", type.asTypeName())
-            parameters.forEachIndexed { i, param ->
+            if (parameters.isNotEmpty()) {
+              add("\n⇥")
+              val isNamedArgumentsSupported = type.toAstClass().primaryConstructor != null
```

Comment:
Curious why you need this? And note: TypeResult.Constructor may not be the primary constructor as you can do:
```kotlin
class Foo() {
   @Inject constructor(bar: Bar) : this()
}
```

Reply:
Ah sorry, meant to leave a comment about that. That was the only way I could figure out to detect if this was a java class or not, because if it is a java class we can't use named parameters.

Reply:
I'd rather detect that some other way if it is possible. If not, I'll just leave a comment explaining why.

Reply:
Oh interesting, yeah this would have to be exposed through the Ast, kapt could check if metadata exists, and ksp should have a method to check if if it's a java class or not. This actually would be necessary as a kotlin class can actually have no primary constructor.

In terms of in the generator, I think it would make sense to have an additional `supportsNamedArguments` field on `TypeResult.Constructor` to make intentions clear and not have to call `toAstClass()` again.

If you don't want to dive that deep into the Ast, I don't mind taking things from here.

Reply:
I just pushed something like that. Let me know if there's a better way to do it.